pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerTimer.java

```diff
@@ -47,7 +47,10 @@ public enum ServerTimer implements AbstractMetrics.Timer {
   SEGMENT_UPLOAD_TIME_MS("milliseconds", false),
 
   TOTAL_CPU_TIME_NS("nanoseconds", false, "Total query cost (thread cpu time + system "
-      + "activities cpu time + response serialization cpu time) for query processing on server.");
+      + "activities cpu time + response serialization cpu time) for query processing on server."),
+
+  EXPIRED_PRIMARY_KEYS_DELETION_TIME_MS("milliseconds", false, "Total time taken to delete expired primary keys based "
```

Comment:
(minor) Suggest renaming it to `UPSERT_REMOVE_EXPIRED_PRIMARY_KEYS_TIME_MS` to match the function name.

As a follow up, we can add timer for other operations as well